File: backend/app/models/enums.py
import enum


class WorkType(str, enum.Enum):
    PLUMBER = "plumber"
    ELECTRICIAN = "electrician"
    CARPENTER = "carpenter"
    IT = "it"


class RequestStatus(str, enum.Enum):
    OPEN = "open"
    OFFER_SELECTED = "offer_selected"
    CANCELLED = "cancelled"


class OfferStatus(str, enum.Enum):
    PENDING = "pending"
    CHOSEN = "chosen"
    REJECTED = "rejected"


class JobStatus(str, enum.Enum):
    PENDING = "pending"
    STARTED = "started"
    FINISHED = "finished"
    CANCELED = "canceled"


class PaymentMethod(str, enum.Enum):
    ONLINE = "online"
    CASH = "cash"


class PaymentStatus(str, enum.Enum):
    PENDING = "pending"
    PAID = "paid"
    FAILED = "failed"


class CanceledBy(str, enum.Enum):
    USER = "user"
    WORKER = "worker"

File: frontend/src/app/core/models/enums.ts
// ============================================================
//  Handy — Core Enums
// ============================================================

/** Work type a handyman specializes in */
export enum WorkType {
  Plumber     = 'plumber',
  Electrician = 'electrician',
  Carpenter   = 'carpenter',
  IT          = 'it',
}

/**
 * Human-readable labels for WorkType are intentionally NOT defined here as plain
 * string maps — all display of a WorkType in the UI must go through the translate
 * pipe with keys 'WORK_TYPE.<value>' (see assets/i18n/en.json / ar.json) so labels
 * localize correctly. Do not reintroduce a hardcoded English/Arabic label map.
 */

/** Job lifecycle states — driven by the user */
export enum JobStatus {
  Pending  = 'pending',
  Started  = 'started',
  Finished = 'finished',
  Canceled = 'canceled',
}

/** How the user paid when marking a job Finished */
export enum PaymentType {
  Online = 'online',
  Cash   = 'cash',
}

/** User roles stored in the JWT */
export enum UserRole {
  User   = 'user',
  Worker = 'worker',
}

/** Account identifier type for login */
export enum IdentifierType {
  Username = 'username',
  Phone    = 'phone',
}

/** All supported worker types as an array (for dropdowns) */
export const WORK_TYPES: WorkType[] = Object.values(WorkType);
